src/main/java/io/lettuce/core/ReadFromImpl.java

```diff
@@ -129,6 +135,200 @@ public ReadFromAnyReplica() {
 
     }
 
+    /**
+     * Read from any node in the subnets. This class does not provide DNS resolution and supports only IP address style
+     * {@link RedisURI} i.e. unavailable when using {@link io.lettuce.core.masterreplica.MasterReplica} with static setup
+     * (provided hosts) and Redis Sentinel with {@literal announce-hostname yes}. Both IPv4 and IPv6 style subnets are supported
+     * but they never match with IP addresses of different version.
+     *
+     * @since x.x.x
+     */
+    static final class ReadFromSubnet extends ReadFrom {
+
+        private final List<SubnetRule> rules = new ArrayList<>();
+
+        /**
+         * @param cidrNotations CIDR-block notation strings, e.g., "192.168.0.0/16" or "2001:db8:abcd:0000::/52".
+         */
+        ReadFromSubnet(String... cidrNotations) {
+            LettuceAssert.notEmpty(cidrNotations, "cidrNotations must not be empty");
+
+            for (String cidrNotation : cidrNotations) {
+                rules.add(createSubnetRule(cidrNotation));
+            }
+        }
+
+        @Override
+        public List<RedisNodeDescription> select(Nodes nodes) {
+            List<RedisNodeDescription> result = new ArrayList<>(nodes.getNodes().size());
+            for (RedisNodeDescription node : nodes) {
+                for (SubnetRule rule : rules) {
+                    if (rule.isInSubnet(node.getUri().getHost())) {
+                        result.add(node);
+                        break;
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        interface SubnetRule {
+
+            boolean isInSubnet(String ipAddress);
+
+        }
+
+        static SubnetRule createSubnetRule(String cidrNotation) {
+            String[] parts = cidrNotation.split("/");
+            LettuceAssert.isTrue(parts.length == 2, "cidrNotation must have exact one '/'");
+
+            String ipAddress = parts[0];
+            int cidrPrefix = Integer.parseInt(parts[1]);
+
+            if (NetUtil.isValidIpV4Address(ipAddress)) {
+                return new Ipv4SubnetRule(ipAddress, cidrPrefix);
+            } else if (NetUtil.isValidIpV6Address(ipAddress)) {
+                return new Ipv6SubnetRule(ipAddress, cidrPrefix);
+            } else {
+                throw new IllegalArgumentException("invalid cidrNotation. cidrNotation=" + cidrNotation);
+            }
+        }
+
+        static class Ipv4SubnetRule implements SubnetRule {
```

Comment:
To avoid license issue (though netty is Apache license),  IPv4/IPv6 rules are re-implemented by myself, so some detailed implementations are different from Netty's one.